lib/Types/MIDI.pm

```diff
@@ -0,0 +1,83 @@
+package Types::MIDI;
+
+# ABSTRACT: Type library for MIDI
+
+use strict;
+use warnings;
+
+use Type::Library
+    -extends => [ qw(
+        Types::Common::Numeric
+    ) ],
+    -declare => qw(
+        Channel
+        Velocity
+        Note
+        PercussionNote
+    );
+use Type::Utils -all;
+
+## no critic (ValuesAndExpressions::ProhibitMagicNumbers)
+
+=type Channel
+
+An L<integer from|Types::Common::Numeric/Types> 0 to 15 corresponding
+to a L<MIDI Channel|/"SEE ALSO">.
+
+=cut
+
+declare Channel, as IntRange [ 0, 15 ];
+
+=type Velocity
+
+An L<integer from|Types::Common::Numeric/Types> 0 to 127 corresponding
+to a L<MIDI Velocity|/"SEE ALSO">.
+
+=cut
+
+declare Velocity, as IntRange [ 0, 127 ];
+
+=type Note
+
+An L<integer from|Types::Common::Numeric/Types> 0 to 127 corresponding
+to a L<MIDI Note Number|/"SEE ALSO">.
+
+=cut
+
+declare Note, as IntRange [ 0, 127 ];
+
+=type PercussionNote
+
+A L</Note> from 27 through 87, corresponding to a value in the
+L<General MIDI 2 Percussion Sound Set|/"SEE ALSO">.
+
+=cut
+
+# TODO: update MIDI-Perl's %MIDI::notenum2percussion with all GM2 sounds
```

Comment:
I doubt DCONKLIN will be putting Level2 stuff into the MIDI module. Maybe you know something I don't, however.

Reply:
> I doubt DCONKLIN will be putting Level2 stuff into the MIDI module. Maybe you know something I don't, however.

[I submitted a patch](https://rt.cpan.org/Ticket/Display.html?id=163587). We’ll see.

Reply:
@mjgardner  May The Force be with you!